src/networking.c

```diff
@@ -1546,6 +1546,9 @@ void clearClientConnectionState(client *c) {
     pubsubUnsubscribeAllChannels(c,0);
     pubsubUnsubscribeShardAllChannels(c, 0);
     pubsubUnsubscribeAllPatterns(c,0);
+    if (c->flags & CLIENT_PUBSUB) {
```

Comment:
it's better using `unmarkClientAsPubSub` here, and remove `CLIENT_PUBSUB` from "clear state" step.